src/app/earn/provider.tsx

```diff
@@ -0,0 +1,512 @@
+  indexTokens: Token[]
+  apy: number | null
+  apy7d: number | null
+  apy30d: number | null
+  nav: number | null
+  tvl: number | null
+  inputToken: Token
+  outputToken: Token
+  inputTokenAmount: bigint
+  inputTokens: Token[]
+  outputTokens: Token[]
+  isFetchingQuote: boolean
+  isFetchingStats: boolean
+  quoteResult: QuoteResult | null
+  stats: BaseTokenStats | null
+  transactionReview: TransactionReview | null
+  onChangeInputTokenAmount: (input: string) => void
+  onSelectInputToken: (tokenSymbol: string) => void
+  onSelectOutputToken: (tokenSymbol: string) => void
+  reset: () => void
+  toggleIsMinting: () => void
+}
+
+export const EarnContext = createContext<Context>({
+  inputValue: '',
+  isMinting: true,
+  balances: [],
+  indexToken: HighYieldETHIndex,
+  indexTokens: [],
+  apy: null,
+  apy7d: null,
+  apy30d: null,
+  nav: null,
+  tvl: null,
+  inputToken: ETH,
+  outputToken: HighYieldETHIndex,
+  inputTokenAmount: BigInt(0),
+  inputTokens: [],
+  outputTokens: [],
+  isFetchingQuote: false,
+  isFetchingStats: true,
+  quoteResult: null,
+  stats: null,
+  transactionReview: null,
+  onChangeInputTokenAmount: () => {},
+  onSelectInputToken: () => {},
+  onSelectOutputToken: () => {},
+  reset: () => {},
+  toggleIsMinting: () => {},
+})
+
+export const useEarnContext = () => useContext(EarnContext)
+
+const defaultParams = {
+  isMinting: true,
+  inputToken: ETH,
+  outputToken: HighYieldETHIndex,
+}
+
+export function EarnProvider(props: { children: any }) {
+  const publicClient = usePublicClient()
+  const { chainId: chainIdRaw, switchChain } = useNetwork()
+  const nativeTokenPrice = useNativeTokenPrice(chainIdRaw)
+  const { address, provider, rpcUrl } = useWallet()
+  const {
+    queryParams: {
+      queryNetwork,
+      queryInputToken,
+      queryOutputToken,
+      queryIsMinting,
+    },
+    updateQueryParams,
+  } = useQueryParams({ ...defaultParams, network: chainIdRaw })
+
+  const [inputValue, setInputValue] = useState('')
+  const [isFetchingQuote, setFetchingQuote] = useState(false)
+  const [isMinting, setMinting] = useState<boolean>(queryIsMinting)
+  const [inputToken, setInputToken] = useState<Token>(queryInputToken)
+  const [outputToken, setOutputToken] = useState<Token>(queryOutputToken)
+  const [flashmintQuote, setFlashmintQuote] = useState<Quote | null>(null)
+  const [swapQuote, setSwapQuote] = useState<Quote | null>(null)
+  const [quoteResult, setQuoteResult] = useState<QuoteResult>({
+    type: QuoteType.flashmint,
+    isAvailable: true,
+    quote: null,
+    error: null,
+  })
+
+  const chainId = useMemo(() => {
+    return chainIdRaw ?? MAINNET.chainId
+  }, [chainIdRaw])
+
+  useEffect(() => {
+    // queryNetwork is only set on the initial load
+    if (queryNetwork) {
+      switchChain({ chainId: queryNetwork })
+    }
+  }, [queryNetwork, switchChain])
+
+  const indexToken = useMemo(() => {
+    if (isMinting) return outputToken
+    return inputToken
```

Comment:
yeah the 'widget logic' on the earn section has not been fully implemented yet, but this is something i'll definitely investigate as part of the next phase

I went ahead and added a redirect so if someone tries to land on `/earn` it will take them to `/swap` as this code isn't ready yet to be seen (someone would need to know the route exists, but figured this was safest)